CMakeLists.txt

```diff
@@ -0,0 +1,102 @@
+cmake_minimum_required(VERSION 3.12)
+
+# Ensure built-in policies from CMake are used, (e.g. improved policies for macOS)
+cmake_policy(VERSION ${CMAKE_MAJOR_VERSION}.${CMAKE_MINOR_VERSION})
+
+project(popt
+	VERSION 1.19
+	DESCRIPTION "Portable library for parsing command line parameters"
+	LANGUAGES C
+)
+
+# Set soversion
+set(POPT_SOVERSION 0)
+
+# Set up GNU conventions and standard FHS paths
+include(GNUInstallDirs)
+
+# Activate CMake package configuration helpers
+include(CMakePackageConfigHelpers)
+
+# set the minimum C standard
+set(CMAKE_C_STANDARD 99)
+set(CMAKE_C_EXTENSIONS ON)
+
+# Build Types
+set(CMAKE_BUILD_TYPE ${CMAKE_BUILD_TYPE}
+	CACHE STRING "Choose the type of build, options are: None Debug Release RelWithDebInfo MinSizeRel SanitizeAddress"
+	FORCE
+)
+
+# AddressSanitize
+set(CMAKE_C_FLAGS_SANITIZEADDRESS
+	"-O2 -g -Wall -fsanitize=address -fno-optimize-sibling-calls -fsanitize-address-use-after-scope -fno-omit-frame-pointer"
```

Comment:
Warning level flags do not belong here. For address sanitation -O2 seems pretty non-optimal in the first place, -fsanitize=address docs explicitly list "-O0, -O1, or -Og" for producing more accurate traces. Moreover, as a developer I want to make my own choices on matters like this, I think address sanitizer option should enable *just that* and not muck with any other compiler flags.

Again I'd leave this out for the initial merge, we can argue the fine details in a separate PR to add this extra functionality.

Reply:
I forgot to drop the `-O2` actually... But I dropped all of it instead...